File: scripts/collect-coverage-data/models/repository.py
from pydantic import BaseModel
from bson import ObjectId

from ...utils import get_uuid, get_time

__all__ = ("Repository",)


class Repository(BaseModel):
    _id: ObjectId
    name: str
    language: str
    default_branch: str
    has_codecov: int
    has_coveralls: int
    has_codeclimate: int
    created_at: int = get_time()
    updated_at: int = get_time()

File: server/models/Repository.js
const mongoose = require('mongoose');

const repositorySchema = new mongoose.Schema({
    name: String,
    language: String,
    default_branch: String,
    has_codecov: Number,
    has_coveralls: Number,
    has_codeclimate: Number,
    created_at: Number,
    updated_at: Number
}, {collection: 'repository'});

module.exports = mongoose.model('Repository', repositorySchema);
